### DISCUSSION TITLE
Customise StreamRedirector

### DISCUSSION THREAD

**Discussion by Leonardo1406** [nvaccess/nvda#15081] [Q&A]
Hello! I have a problem, I want to modify the StreamRedirector class to add the attribute that calls f.fileno() so that NVDA's log viewer recognizes that fileno() is an attribute that belongs to the StreamRedirector class. But I can't do that within the NVDA environment...

**Comment by josephsl** [comment]
Hi, please tell us more, including use cases and possible benefits. Thanks.

From: Leonardo1406 ***@***.***> 
Sent: Friday, June 30, 2023 10:51 AM
To: nvaccess/nvda ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [nvaccess/nvda] Customise StreamRedirector (Discussion #15081)

 

Hello! I have a problem, I want to modify the StreamRedirector class to add the attribute that calls f.fileno() so that NVDA's log viewer recognizes that fileno() is an attribute that belongs to the StreamRedirector class. But I can't do that within the NVDA environment...

**Reply by Leonardo1406** [reply]
Hello!
I'm sorry for answering so late.
I actually want to modify the class just for an add-on I'm developing, without affecting NVDA's current behavior.
Is there a procedure I need to follow? Should I enable scratchpad code loading, for example?
It's just that I was told that I don't need to activate the load code to develop an add-on.
But is there any NVDA configuration that I need to modify for my plugin to work? Or do I simply need to insert additional logic into it?
Thanks
